api/utils/db.py

```diff
@@ -59,6 +60,12 @@ def authenticate(
     If a token (OR Google IAP auth jwt) is provided,
     return the email, else raise an Exception
     """
+
+    if default_user := get_default_user():
```

Comment:
Yeah I don't love it either, the problem is that the generated API by default will get your local GCP credentials. So when running scripts like `generate_data.py`, `x_goog_iap_jwt_assertion` will exist and the default user won't be used, this causes problems because the authenticated user doesn't have permissions. I think it is pretty safe as `get_default_user()` has an explicit environment check.

Reply:
It's creeping outside the scope a bit, but I wonder if we should change it in the generated API to not provide an auth token if local / test? Then we wouldn't need to re-arrange here, and would probably speed up local testing by not always generating this token which does take some amount of time.

https://github.com/populationgenomics/metamist/blob/846c6b80006e0264c1153998bb825f0804bd68c9/openapi-templates/configuration.mustache#L502-L503

Reply:
Just checked this and as long as you have the SM_ENVIRONMENT=local set when running the generate data script it works fine, I must have not had it set when I was testing. I've reordered the checks back to what they were